### ISSUE TITLE
For the rmse of odom result

### ISSUE THREAD

**Issue by bbbrian94** [Huangying-Zhan/Depth-VO-Feat#28]
After reading your paper's experiment part, I updated the data_builder(which include only 00-08's 19600 pairs of images) and retrain the both part of Depth and Depth+Odom. Finally, my trained model got some huge difference from your Temporal.model.
"
Sequence: 9
Average translational RMSE (%):  18.29599141944674
Average rotational error (deg/100m):  6.84948743104451
Sequence: 10
Average translational RMSE (%):  31.608499883939217
Average rotational error (deg/100m):  13.520508069343077
"
compare to yours:
"
Sequence: 9
Average translational RMSE (%):  11.260165600943994
Average rotational error (deg/100m):  3.8829374695264844
Sequence: 10
Average translational RMSE (%):  10.655759483817
Average rotational error (deg/100m):  3.944550738667936
"

Could you help me to figure out why there is huge difference? 

BTW, is your "Baseline.caffemodel" trained by the seq00-09 or trained by raw data's 23455 pairs? 

can I train my own Temporal.caffemodel from Baseline.caffemodel?

If possible would you like to share any train.protxt for the depth/odom trainning regarding to your paper says you are following T.Zhou's paper to finish that part?

Also could you help to explain why you choose to use R12 to train odometry, but when you test with eval_tools.py, you are using Left image "image_02"?

Thank you!
my email:shaofeng@umich.edu

**Comment by Huangying-Zhan** [comment]
@bbbrian94 Thanks for your interest in our work.
I am not sure about the cause of the huge difference you got. However, I am happy to discuss and help.

(1) is your "Baseline.caffemodel" trained by the seq00-09 or trained by raw data's 23455 pairs?
Depends on the evaluation task, for depth estimation evaluation, I used raw data (i.e. eigen split) to train the network.
For odometry evaluation, I used odometry split seq00-08 for training.
The reason is mentioned in the paper, basically, there is an overlap between train set and test set of Eigen split and Odom split.

(2) can I train my own Temporal.caffemodel from Baseline.caffemodel?
Yes, as mentioned in the project instruction, I used Baseline model to initialize Temporal experiment. 
I guess you might train the Temporal model from scratch. You may try with this initialization. 

(3) could you help to explain why you choose to use R12 to train odometry, but when you test with eval_tools.py, you are using Left image "image_02"?
In the experiments, I chose to predict depth on R2 (no specific reason, you may choose L2 as well) and use it as a reference view. However, in the evaluation, I test with left images because of following previous work (fair comparison).

Hope it helps and let me know if there is any problem.

**Comment by bbbrian94** [comment]
> @bbbrian94 Thanks for your interest in our work.
> I am not sure about the cause of the huge difference you got. However, I am happy to discuss and help.
> 
> (1) is your "Baseline.caffemodel" trained by the seq00-09 or trained by raw data's 23455 pairs?
> Depends on the evaluation task, for depth estimation evaluation, I used raw data (i.e. eigen split) to train the network.
> For odometry evaluation, I used odometry split seq00-08 for training.
> The reason is mentioned in the paper, basically, there is an overlap between train set and test set of Eigen split and Odom split.
> 
> (2) can I train my own Temporal.caffemodel from Baseline.caffemodel?
> Yes, as mentioned in the project instruction, I used Baseline model to initialize Temporal experiment.
> I guess you might train the Temporal model from scratch. You may try with this initialization.
> 
> (3) could you help to explain why you choose to use R12 to train odometry, but when you test with eval_tools.py, you are using Left image "image_02"?
> In the experiments, I chose to predict depth on R2 (no specific reason, you may choose L2 as well) and use it as a reference view. However, in the evaluation, I test with left images because of following previous work (fair comparison).
> 
> Hope it helps and let me know if there is any problem.

Really appreciate for your reply!

For the second part, I mean, from your given trained model Baseline.caffemodel (I guess you got this by using 23455 raw data), Can i train my own Temporal model based on sequence odom split (using 19600's data)? Based on your paper, I may not do that. the result I post here is based on depth training on raw data, then I train once again on Odom split data to get Temporal.caffemodel Odom+ depth. I will try to train both Baseline and Temporal with same odom split data (sequence 19600's).
 

BTW, All the given code here on github is based on raw data. I am not sure whether your baseline.caffemodel is also based on raw data or not. Under the repository of the trained model of Odom split, I cannot find the similar caffemodel such as Baseline. if you give one, that might save a lot of training time for us ^_^  

Finally, your given Baseline model in the eigen split repo is trained by raw data, right?  Your given Temporal model in odom split is trained based on Odom split's baseline model + odom split data, right?

Thanks in advance!

**Comment by Huangying-Zhan** [comment]
For the released model, I didn't release Baseline (Odom split) since it is not for odometry evaluation. I have uploaded the Baseline model trained from Odom split. Please check the dropbox.

your given Baseline model in the eigen split repo is trained by raw data, right? Your given Temporal model in odom split is trained based on Odom split's baseline model + odom split data, right?
Yes, models in eigen_split are trained using raw data while models in odometry_split are trained using odom split data.

**Comment by bbbrian94** [comment]
Thanks a lot for that baseline model! 

Just the last Question before I close this issue. your data builder for the odom split and the eigen split is very similar. The only difference is at the data input, one is raw data, another is sequence data, right? and all the prototxt for training is also same, right? the learning rate is still at 0.001 but not T.Zhou's 0.0002, right?
